atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/TransactionManagerTest.java

```diff
@@ -168,6 +170,74 @@ public void shouldNotConflictIfImmutableTimestampLockExpiresIfNoReadsOrWrites()
         txnManagerWithMocks.runTaskThrowOnConflict(txn -> null);
     }
 
+    @Test
+    public void writeDoesNotPopulateTimestampCache() {
+        Transaction transaction = txMgr.runTaskWithRetry(txn -> {
+            put(txn, TEST_TABLE, "row", "column", "test");
+            return txn;
+        });
```

Comment:
Might be cleaner in terms of minimum visibility to just return the timestamp from the transaction task.